visma/matrix/structure.py

```diff
@@ -65,6 +65,37 @@ def isSquare(self):
         else:
             return False
 
+    def isIdentity(self):
+        """Checks if matrix is identity
+
+        Returns:
+            bool -- if identity matrix or not
+        """
+        if self.isDiagonal():
+            for i in range(0, self.dim[0]):
+                if self.value[i][i][0].__str__() != "{1.0}":
```

Comment:
The matrix `[1 + x, 0 + y ; 0 , 1]` will be identified as both Diagonal and Identity matrix.

Reply:
will checking for length of arguments suffice @8hantanu ?